### PR TITLE
fix: Sanitise toast error msgs

### LINKED ISSUE #22318: [Bug]-[530]:Sanitise error messages on toast alerts
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Remove the error type(`UncaughtPromiseRejection`, `ReferenceErrror`, `TypeError`, etc...) from all error messages displayed on toast alerts. 

### Steps To Reproduce

1. Drag and drop a button
2. Select onClick action to Navigate To
3. Specify an invalid URL
4. Click the button and notice the error type(`UncaughtPromiseRejection`) displayed on the toast alert

uncaughtpromiserejection

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by rimildeyjsr**
## Description

Currently, the error messages in the toasts contain the names of the errors (like Reference error, uncaught promise rejection error, etc.,). These are unhelpful to users (especially if they are not programmers) and do not convey any actionable feedback to the user who is trying to fix and debug the app.

You can see it in action [here](https://www.loom.com/share/e946f779dd1147f38eec1588a84821b2).

This PR aims to remove the names of these errors from the toast messages so that the action to fix them can be highlighted. We are retaining the names of the errors for the console, so that programmers using the console, can get a full context of the error.

Fixes #22318

Media

Previous behavior - https://www.loom.com/share/e946f779dd1147f38eec1588a84821b2

Current behavior - https://www.loom.com/share/83fd8d08ed114f8b830acadb9894e4b1

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Jest
- Cypress

### Test Plan
- Reference error check
- Uncaught promise rejection check

### Issues raised during DP testing
- none

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer-reviewed by QA
- [x] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [x] Added Test Plan Approved label after reviewing all Cypress test

### CODE REVIEW

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

File: app/client/src/workers/Evaluation/evaluate.ts [hunk 1]
```diff
@@ -338,3 +344,15 @@ export async function evaluateAsync(
...
+
+export function convertAllDataTypesToString(e: any): string {
+  // Functions do not get converted properly with JSON.stringify
+  // So using String fot functions
+  // Types like [], {} get converted to "" using String
+  // hence using JSON.stringify for the rest
+  if (typeof e === "function") {
+    return String(e);
+  } else {
+    return JSON.stringify(e);
```
Comment (arunvjn) [inline app/client/src/workers/Evaluation/evaluate.ts:356]: Can we wrap this inside a try catch block ? JSON.stringify could throw errors for cyclical references

**Review by arunvjn — COMMENTED** [review]
Added some minor comments. Rest looks good!

File: app/client/src/sagas/EvaluationsSaga.ts [hunk 2]
```diff
@@ -332,17 +329,8 @@ export function* evaluateAndExecuteDynamicTrigger(
...
-    if (
-      errors[0].errorMessage !==
-      "UncaughtPromiseRejection: User cancelled action execution"
-    ) {
-      const errorMessage =
-        `${errors[0].errorMessage.name}: ${errors[0].errorMessage.message}` ||
-        errors[0].message;
-      throw new UncaughtPromiseError(errorMessage);
-    }
+  if (errors.length > 0) {
```
[the remark this answers is not in the document]
Comment (rimildeyjsr) [reply app/client/src/sagas/EvaluationsSaga.ts:332]: done

File: app/client/src/workers/Evaluation/evaluate.ts (hunk 1 above)
Comment (rimildeyjsr) [reply app/client/src/workers/Evaluation/evaluate.ts:356]: done

**Review by arunvjn — DISMISSED** [review]

**Comment by ramsaptami** [comment]
LGTM
cc: @rimildeyjsr

**Review by arunvjn — DISMISSED** [review]

File: app/client/src/sagas/ActionExecution/geolocationSaga.ts [hunk 3]
```diff
@@ -75,11 +75,7 @@ export const extractGeoLocation = (
  * Hence we're creating a new object with same structure which can be passed to the worker thread
  */
 function sanitizeGeolocationError(error: GeolocationPositionError) {
-  const { code, message } = error;
-  return {
-    code,
-    message,
-  };
+  return error.message;
```
Comment (rimildeyjsr) [inline app/client/src/sagas/ActionExecution/geolocationSaga.ts:78]: @arunvjn - is this change ok to make?

Comment (arunvjn) [reply app/client/src/sagas/ActionExecution/geolocationSaga.ts:78]: Callback expects the error object to be in the format `{ code, message }`. This change will cause user apps to break.

**Review by arunvjn — APPROVED** [review]

**Pull request merged**